### PR TITLE
Add manifest for Tambourine

### PR DESCRIPTION
**Pull request by MMarco94**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)* Explanation added
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

Explanation for permissions:
- `--socket=session-bus`: the app is written in Java, and interacting with the DBus has issue that I haven't solved yet. In the meantime, I'm granting this permission. I need DBus access for the MPRIS protocol.
- `--filesystem=host:ro`: for similar reasons as above, I cannot manage to connect to the File Chooser portal using the DBus. In the meantime, I'm granting this permission. I need to access the file in order to reproduce the songs.

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: io.github.mmarco94.tambourine.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --share=ipc
+  - --device=dri
+  - --env=PATH=/app/jre/bin:/app/bin:/usr/bin
+  - --env=JAVA_HOME=/app/jre
+  # TODO: this shouldn't be used, but if I don't do this, the DBUS connection gets closed
+  # almost immediately because of "Invalid client serial"
+  - --socket=session-bus
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.yml:58]: you should determine which name you need. This is NOT allowed.

File: io.github.mmarco94.tambourine.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+cleanup-commands:
+  - mkdir -p /app/lib/ffmpeg
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.yml:60]: this should be in the post install somewhere. This is NOT a cleanup.

File: run.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+set -ex
+
+java \
```
Comment (hfiguiere) [inline run.sh:5]:
```suggestion
exec java \
```

File: io.github.mmarco94.tambourine.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+modules:
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk17/install.sh
+  - name: tambourine
+    buildsystem: simple
+    build-commands:
+      # ffsampledsp requires libbz2.so.1.0, however Flatpak only provides libbz2.so.1.0.8
+      - ln -s /usr/lib/x86_64-linux-gnu/libbz2.so.1.0.8 /app/jre/lib/libbz2.so.1.0
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.yml:22]: how is this gonna work on `aarch64`?

Comment (hfiguiere) [reply io.github.mmarco94.tambourine.yml:22]: also this is bound to fail, there is a reason with soname have versions.

File: io.github.mmarco94.tambourine.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      - # Installing the actual app
+      - install -D tambourine-linux-x64-1.0.0.jar /app/bin/tambourine.jar
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.yml:24]: so you build an aarch64 package, still

isn't Java supposed to run anywhere?

File: io.github.mmarco94.tambourine.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    </description>
+    <url type="homepage">https://github.com/MMarco94/tambourine-music-player/</url>
+    <url type="bugtracker">https://github.com/MMarco94/tambourine-music-player/issues</url>
+    <launchable type="desktop-id">io.github.mmarco94.tambourine.desktop</launchable>
+    <project_license>GPL-3.0</project_license>
+    <developer_name>Marco Marangoni</developer_name>
+    <categories>
+        <category>Audio</category>
+    </categories>
+    <content_rating type="oars-1.0" />
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.metainfo.xml:35]:
```suggestion
    <content_rating type="oars-1.1" />
```

File: io.github.mmarco94.tambourine.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        <screenshot>
+            <image type="source" width="1552" height="1089">
+                https://raw.githubusercontent.com/MMarco94/tambourine-music-player/main/screenshots/s4.png
+            </image>
+        </screenshot>
+    </screenshots>
+    <recommends>
+        <control>pointing</control>
+    </recommends>
+</component>
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.metainfo.xml:66]: missing required `releases`

File: io.github.mmarco94.tambourine.desktop [hunk 8]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Categories=Audio;
+Exec=run.sh
+Icon=io.github.mmarco94.tambourine
+Name=Tambourine Music Player
+Comment=A simple music player for your library
+Type=Application
+X-Flatpak=io.github.mmarco94.tambourine
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.desktop:8]:
```suggestion
```

File: io.github.mmarco94.tambourine.yml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+runtime-version: '22.08'
+add-extensions:
+  org.freedesktop.Platform.ffmpeg-full:
+    directory: lib/ffmpeg
+    version: '22.08'
+    add-ld-path: .
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+command: /app/bin/run.sh
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.yml:12]:
```suggestion
command: run.sh
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]
the icon, .desktop and appstream files should be submitted upstream for inclusion. (none are specific to flatpak)

File: io.github.mmarco94.tambourine.yml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+      - install -D io.github.mmarco94.tambourine.metainfo.xml /app/share/metainfo/io.github.mmarco94.tambourine.metainfo.xml
+      - install -D icon.png /app/share/icons/hicolor/256x256/apps/io.github.mmarco94.tambourine.png
+    sources:
+      - type: file
+        path: io.github.mmarco94.tambourine.desktop
+      - type: file
+        path: io.github.mmarco94.tambourine.metainfo.xml
+      - type: file
+        path: icon.png
+finish-args:
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.yml:45]: finish args are found ABOVE modules.

File: io.github.mmarco94.tambourine.yml [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+  - --filesystem=xdg-music:ro
+  # TODO: remove once the file portal integration works
+  - --filesystem=host:ro
```
Comment (hfiguiere) [inline io.github.mmarco94.tambourine.yml:48]: this filesystem permission are redundants. and one of the is broad.

File: io.github.mmarco94.tambourine.yml (hunk 4 above)
Comment (MMarco94) [reply io.github.mmarco94.tambourine.yml:22]: > how is this gonna work on aarch64?

It isn't. Unfortunately one of my dependency is only compiled for x86. Is there a way to restrict the manifest to be x86 only?

Comment (MMarco94) [reply io.github.mmarco94.tambourine.yml:22]: > also this is bound to fail, there is a reason with soname have versions.

This is adding a symlink from a more to a less specific `.so` (`1.0.8` -> `1.0`), so it should work

File: io.github.mmarco94.tambourine.yml (hunk 1 above)
Comment (MMarco94) [reply io.github.mmarco94.tambourine.yml:58]: I know which one I need, the problem is that the Flatpak runtime closes the DBus connection with an `Invalid client serial` error after a while.

The only solution I found was to do this.

File: io.github.mmarco94.tambourine.yml (hunk 11 above)
Comment (MMarco94) [reply io.github.mmarco94.tambourine.yml:48]: Like I said in the overview, unfortunately I couldn't get the file portal to work :(

I'm using Java, so there's no "default" integration with the file portal, and I tried to send the command  to the DBus without luck.

Adding `host` access is a compromise while I sort the issue.

File: io.github.mmarco94.tambourine.yml (hunk 4 above)
Comment (hfiguiere) [reply io.github.mmarco94.tambourine.yml:22]: why not building from source instead?

Comment (MMarco94) [reply io.github.mmarco94.tambourine.yml:22]: It's way out of scope for my project. 
I've opened https://github.com/hendriks73/ffsampledsp/issues/17 so they can fix it upstream

File: io.github.mmarco94.tambourine.yml (hunk 11 above)
Comment (MMarco94) [reply io.github.mmarco94.tambourine.yml]: I've opened an issue to get the file picker issues sorted: https://github.com/LWJGL/lwjgl3/issues/893

File: io.github.mmarco94.tambourine.yml (hunk 1 above)
Comment (MMarco94) [reply io.github.mmarco94.tambourine.yml]: I've also opened https://github.com/hypfvieh/dbus-java/issues/220 to get my issue with the dbus fixed

**Comment by MMarco94** [comment]
I fixed the issues I could solve, and opened bug reports for the workarounds I had to make due to my dependencies.

How can we get this merged? 

I can compromise on the file system access for the time being: if an user wants to open files outside the music folder they can use Flatseal I guess. That's of course not okay, but for a first release it's acceptable IMHO.

What do you think?

**Comment by barthalion** [comment]
You need to disable the aarch64 build: https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

**Comment by barthalion** [comment]
bot, build io.github.mmarco94.tambourine

File: io.github.mmarco94.tambourine.yml [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --share=ipc
+  - --device=dri
+  - --env=PATH=/app/jre/bin:/app/bin:/usr/bin
+  - --env=JAVA_HOME=/app/jre
+  # TODO: this shouldn't be used, but if I don't do this, the DBUS connection gets closed
+  # almost immediately because of "Invalid client serial"
+  - --socket=session-bus
```
Comment (barthalion) [inline io.github.mmarco94.tambourine.yml:25]: Did you check what's going on with `--log-session-bus`? This is a huge blocker.

Comment (MMarco94) [reply io.github.mmarco94.tambourine.yml]: Yes :)

I found a bug in the library I use for accessing the DBus, see https://github.com/hypfvieh/dbus-java/issues/220

Now I have removed that permission

**Comment by MMarco94** [comment]
bot, build io.github.mmarco94.tambourine

**Comment by MMarco94** [comment]
bot, build io.github.mmarco94.tambourine

File: io.github.mmarco94.tambourine.yml [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+command: run.sh
+finish-args:
+  - --filesystem=xdg-run/gvfs
+  - --filesystem=xdg-run/gvfsd
+  # Workaround until https://github.com/LWJGL/lwjgl3/issues/893 is fixed
+  - --filesystem=host:ro
+  - --socket=wayland
+  - --socket=fallback-x11
```
Comment (barthalion) [inline io.github.mmarco94.tambourine.yml:19]:
```suggestion
  - --socket=x11
```

It doesn't work on Wayland.

Comment (MMarco94) [reply io.github.mmarco94.tambourine.yml]: Done. I've added https://github.com/MMarco94/tambourine-music-player/issues/3 to investigate why it doesn't work

**Comment by MMarco94** [comment]
bot, build io.github.mmarco94.tambourine

**Comment by MMarco94** [comment]
Thanks for your time for the review! 
In the latest version, I implemented all the suggested changes, except for the file permissions, as there's no way I could find to connect to the File portal in Java, see https://github.com/LWJGL/lwjgl3/issues/893#issue-1764307305. 

Once a separate issue with the DBus is fixed (https://github.com/hypfvieh/dbus-java/issues/220), I'll try to connect to the file portal directly.

Do you think this issue is a blocker, or can the initial version be published?

Thanks again for your time, it's really appreciated!

**Comment by barthalion** [comment]
bot, build io.github.mmarco94.tambourine

**Comment by barthalion** [comment]
For some reason, it doesn't actually draw any window on Wayland session, even after recent changes.

```
~ 4.2s ❱ flatpak run io.github.mmarco94.tambourine
+ exec java -Djava.util.prefs.userRoot=/var/home/bpiotrowski/.var/app/io.github.mmarco94.tambourine/config -Djava.io.tmpdir=/var/home/bpiotrowski/.var/app/io.github.mmarco94.tambourine/cache/tmp/ -jar /app/bin/tambourine.jar
SLF4J: A number (58) of logging calls during the initialization phase have been intercepted and are
SLF4J: now being replayed. These are subject to the filtering rules of the underlying logging system.
SLF4J: See also https://www.slf4j.org/codes.html#replay
2023-06-26 11:45:39.459 |  INFO | org.freedesktop.dbus.connections.transports.TransportBuilder | Using transport dbus-java-transport-native-unixsocket for address unix:path=/run/flatpak/bus
2023-06-26 11:45:39.465 | DEBUG | io.github.mmarco94.tambourine.data.Library | Loading songs took 2ms
2023-06-26 11:45:39.466 | DEBUG | io.github.mmarco94.tambourine.data.Library | Creating library took 1ms
Exception in thread "main" java.lang.ExceptionInInitializerError
	at androidx.compose.ui.awt.ComposeLayer.<init>(ComposeLayer.desktop.kt:92)
	at androidx.compose.ui.awt.ComposeWindowDelegate.<init>(ComposeWindowDelegate.desktop.kt:60)
	at androidx.compose.ui.awt.ComposeWindow.<init>(ComposeWindow.desktop.kt:61)
	at androidx.compose.ui.awt.ComposeWindow.<init>(ComposeWindow.desktop.kt:59)
	at androidx.compose.ui.window.Window_desktopKt$Window$3.invoke(Window.desktop.kt:178)
	at androidx.compose.ui.window.Window_desktopKt$Window$3.invoke(Window.desktop.kt:172)
	at androidx.compose.ui.window.Window_desktopKt$Window$10$1.invoke(Window.desktop.kt:403)
	at androidx.compose.ui.window.Window_desktopKt$Window$10$1.invoke(Window.desktop.kt:402)
	at androidx.compose.ui.window.AwtWindow_desktopKt$AwtWindow$2.invoke(AwtWindow.desktop.kt:74)
	at androidx.compose.ui.window.AwtWindow_desktopKt$AwtWindow$2.invoke(AwtWindow.desktop.kt:73)
	at androidx.compose.runtime.DisposableEffectImpl.onRemembered(Effects.kt:81)
	at androidx.compose.runtime.CompositionImpl$RememberEventDispatcher.dispatchRememberObservers(Composition.kt:1105)
	at androidx.compose.runtime.CompositionImpl.applyChangesInLocked(Composition.kt:820)
	at androidx.compose.runtime.CompositionImpl.applyChanges(Composition.kt:842)
	at androidx.compose.runtime.Recomposer.composeInitial$runtime(Recomposer.kt:994)
	at androidx.compose.runtime.CompositionImpl.setContent(Composition.kt:519)
	at androidx.compose.ui.window.Application_desktopKt$awaitApplication$2$1$2.invokeSuspend(Application.desktop.kt:219)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:771)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:741)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.io.IOException: No such file or directory
	at java.base/java.io.UnixFileSystem.createFileExclusively(Native Method)
	at java.base/java.io.File.createTempFile(File.java:2170)
	at org.jetbrains.skiko.Library.unpackIfNeeded(Library.kt:37)
	at org.jetbrains.skiko.Library.findAndLoad(Library.kt:112)
	at org.jetbrains.skiko.Library.load(Library.kt:59)
	at org.jetbrains.skiko.SkiaLayer.<clinit>(SkiaLayer.awt.kt:32)
	... 32 more
```

**Comment by MMarco94** [comment]
bot, build io.github.mmarco94.tambourine

**Comment by MMarco94** [comment]
bot, build io.github.mmarco94.tambourine

**Comment by MMarco94** [comment]
Thanks for the review. I was able to reproduce the  issue.  I'll fix it shortly.

PS: Sorry, I didn't plan to use you as a beta tester :sweat_smile:

**Comment by MMarco94** [comment]
Ok, the latest version should fix the issue you've faced. Could you give it another shot?

**Comment by barthalion** [comment]
Something didn't work out here. Let's try again.

**Pull request closed without merging**